### PR TITLE
Fluence log

### LINKED ISSUE #419: Software logs readability
Proof read our software logs for readability and consistency. A user should be able to get the idea of what happens inside systems by looking at the logs. Components: Node, worker, vm (?).

### PR DESCRIPTION
**Pull request by alari**
Fixes #419

### CODE REVIEW

File: effects/docker/src/main/scala/fluence/effects/docker/DockerIO.scala [hunk 1]
```diff
@@ -196,54 +180,46 @@ class DockerIO[F[_]: Monad: LiftIO: ContextShift: Defer](
...
-          IO(logger.info(s"disconnecting container ${container.containerId} from network ${network.name} "))
-            .flatMap(_ => runShellVoid(s"docker network disconnect ${network.name} ${container.containerId}"))
-            .handleError {
-              case NonFatal(err) ⇒
-                logger.error(
-                  s"Trying to disconnect container ${container.containerId} from network ${network.name}, got error $err",
-                  err
-                )
-          }
+          Log[F].info(s"disconnecting container ${container.containerId} from network ${network.name} ")
```
Comment (folex) [inline effects/docker/src/main/scala/fluence/effects/docker/DockerIO.scala]: Too bad we can't log container's name here also

File: effects/kvstore/src/main/scala/fluence/effects/kvstore/RocksDBStore.scala [hunk 2]
```diff
@@ -142,28 +145,33 @@ object RocksDBStore extends LazyLogging {
         ctx ⇒ IO(ctx.shutdown()).to[F]
       )
 
-      _ = logger.trace("we have created ctx")
+      _ = Log.resource[F].trace("we have created ctx")
```
Comment (alari) [inline effects/kvstore/src/main/scala/fluence/effects/kvstore/RocksDBStore.scala]: There must be `<-`

File: node/src/main/scala/fluence/node/eth/FluenceContract.scala [hunk 3]
```diff
@@ -160,16 +162,14 @@ class FluenceContract(private[eth] val ethClient: EthClient, private[eth] val co
...
-        Sync[F].delay(logger.info("Getting all previously prepared clusters canceled."))
+        Log[F].info("Getting all previously prepared clusters canceled.")
       case ExitCase.Completed =>
-        Sync[F].delay(logger.info("Got all the previously prepared clusters. Now switching to the new clusters."))
+        Log[F].info("Got all the previously prepared clusters. Now switching to the new clusters.")
       case ExitCase.Error(err) =>
-        Sync[F]
-          .delay(logger.info(s"Error on getting all previously clusters: $err."))
-          .map(_ => err.printStackTrace())
+        Log[F].warn(s"Error on getting all previously clusters: $err.", err)
```
Comment (folex) [inline node/src/main/scala/fluence/node/eth/FluenceContract.scala]: all previously **prepared** clusters

File: node/src/main/scala/fluence/node/workers/WorkersHttp.scala [hunk 4]
```diff
@@ -18,82 +18,95 @@ package fluence.node.workers
...
+              NotFound("App not found on the node")
+
+          case Some(res) ⇒
+            res match {
+              case Right(result) ⇒
+                log.trace(s"RPC responding with OK: $result") *>
+                  Ok(result)
+
+              case Left(RpcRequestFailed(err)) ⇒
+                log.warn(s"RPC request failed: $err", err) *>
```
Comment (folex) [inline node/src/main/scala/fluence/node/workers/WorkersHttp.scala]: error printed twice

File: node/src/main/scala/fluence/node/workers/WorkersHttp.scala [hunk 5]
```diff
@@ -18,82 +18,95 @@ package fluence.node.workers
...
+                  InternalServerError(err.getMessage)
+
+              case Left(err: RpcRequestErrored) ⇒
+                log.warn(s"RPC request errored: $err", err) *>
```
Comment (folex) [inline node/src/main/scala/fluence/node/workers/WorkersHttp.scala]: error printed twice

File: node/src/main/scala/fluence/node/workers/WorkersHttp.scala [hunk 6]
```diff
@@ -18,82 +18,95 @@ package fluence.node.workers
...
+                  InternalServerError(err.error)
+
+              case Left(RpcBodyMalformed(err)) ⇒
+                log.debug(s"RPC body malformed: $err") *>
```
Comment (folex) [inline node/src/main/scala/fluence/node/workers/WorkersHttp.scala]: Hm. I wonder, why is that log a debug one? Looks like a `warn`

File: node/src/main/scala/fluence/node/workers/WorkersHttp.scala [hunk 7]
```diff
@@ -18,82 +18,95 @@ package fluence.node.workers
...
+                log.debug(s"Requested app $appId, but there's no such worker in the pool") *>
+                  NotFound("App not found on the node")
+            }
         }
 
       case req @ POST -> Root / LongVar(appId) / "tx" :? QueryId(id) ⇒
-        req.decode[String] { tx ⇒
-          logger.debug(s"TendermintRpc broadcastTxSync request: \ntx: $tx,\n id: $id")
-          withTendermint(appId)(_.broadcastTxSync(tx, id.getOrElse("dontcare")))
+        LogFactory[F].init("http", "tx") >>= { implicit log ⇒
```
Comment (folex) [inline node/src/main/scala/fluence/node/workers/WorkersHttp.scala]: Maybe also add `tx.id` to the log scope?

File: node/src/main/scala/fluence/node/workers/tendermint/BlockUploading.scala [hunk 8]
```diff
@@ -80,7 +81,7 @@ class BlockUploading[F[_]: ConcurrentEffect: Timer](history: BlockHistory[F]) ex
     Block(blockJson) match {
       case Left(e) =>
         // TODO: load block through TendermintRPC (not WRPC) again
-        Applicative[F].pure(logger.error(s"BlockUploading: app $appId failed to parse Tendermint block: $e"))
+        Log[F].error(s"BlockUploading: app $appId failed to parse Tendermint block: $e", e)
```
Comment (folex) [inline node/src/main/scala/fluence/node/workers/tendermint/BlockUploading.scala]: Wouldn't there be an `appId` in the Log Scope?

File: node/src/main/scala/fluence/node/workers/tendermint/BlockUploading.scala [hunk 9]
```diff
@@ -98,13 +99,10 @@ class BlockUploading[F[_]: ConcurrentEffect: Timer](history: BlockHistory[F]) ex
...
             processF,
             (e: EffectError) =>
-              Applicative[F].pure(
-                logger.error(s"BlockUploading: app $appId error uploading block ${block.header.height}: $e")
-            )
-          )
-          .map(
-            _ => logger.info(s"BlockUploading: app $appId block ${block.header.height} uploaded")
-          )
+              Log[F].error(s"BlockUploading: app $appId error uploading block ${block.header.height}: $e", e)
```
Comment (folex) [inline node/src/main/scala/fluence/node/workers/tendermint/BlockUploading.scala]: error printed twice

File: node/src/main/scala/fluence/node/workers/tendermint/BlockUploading.scala (hunk 8 above)
Comment (folex) [reply node/src/main/scala/fluence/node/workers/tendermint/BlockUploading.scala]: also, error printed twice

**Comment by alari** [comment]
CLI test fails on all branches now, so is unrelated to this PR and must be fixed separately.

**Pull request merged**